src/SharpCompress/Archives/Zip/ZipArchive.cs

```diff
@@ -131,33 +184,41 @@ internal ZipArchive(Stream stream, ReaderOptions readerOptions)
 
         protected override IEnumerable<ZipVolume> LoadVolumes(IEnumerable<Stream> streams)
         {
-            return new ZipVolume(streams.First(), ReaderOptions).AsEnumerable();
+            var st = streams.ToList(); //swap the zip to the end
+            var tmp = st[0];
+            st.RemoveAt(0);
+            st.Add(tmp);
+
+            foreach (var s in st)
+                yield return new ZipVolume(s, base.ReaderOptions);
         }
 
         protected override IEnumerable<ZipArchiveEntry> LoadEntries(IEnumerable<ZipVolume> volumes)
         {
-            var volume = volumes.Single();
-            Stream stream = volume.Stream;
-            foreach (ZipHeader h in headerFactory.ReadSeekableHeader(stream))
+            var vols = volumes.ToArray();
```

Comment:
More eager loading here.  I guess we're saying multi-volume has to have access to all streams/volumes at the start?  RAR doesn't need to operate this way but I guess Zip does because you have to have access to the directory at the end?

Reply:
It needs the first and last item at a minimum. the z01 contains the header, the zip (last part) contains the central directory. Unlike rar there's no header per file. Files that span volumes just continue in to the next file. Each entry has the diskpart number where the file starts.

You could open the next part when it's required like with Rar. This PR could be classed as a rough first effort to demonstrate an understanding of the format, I don't have a good enough understanding of all that the library is capable of to make (or attempt) significant change. It would be good to cater for multi volume and split in one set of logic behind an interface where parts can be loaded as required etc.

Reply:
yeah, I think there's more thought needed but feels like this can be a good first pass